File: js/datachannels/angular/src/app/types/message.ts
export interface Message {
  type: MessageType;
  date: Date;
  text: string;
  messageId?: string;
  to?: string;
  status?: string;
  from?: string;
  isDirect?: boolean;
}

export type MessageType =
  'received_message' | 'sent_message' | 'received_broadcast'

File: android/datachannels/kotlin/app/src/main/java/com/infobip/rtc/showcase/data/Message.kt
package com.infobip.rtc.showcase.data

import java.util.Date

data class Message(
    var type: MessageType,
    var date: Date,
    var text: String,
    var status: MessageStatus,
    var messageId: String?,
    var to: String?,
    var from: String?,
    var isDirect: Boolean?
)

enum class MessageType {
    SENT, RECEIVED, BROADCAST
}

enum class MessageStatus(val value: String) {
    SENT("Sent"), DELIVERED("Delivered")
}